software/firmware/calibrate.py

```diff
@@ -1,41 +1,68 @@
-from europi import AnalogueInput, MAX_UINT16
-from machine import Pin, PWM, freq
+from machine import Pin, ADC, PWM
+from time import sleep
+from europi import oled, b1, b2
 
-# Overclock for faster calibration
-freq(250_000_000)
+machine.freq(250000000)
 
-ain = AnalogueInput(26)
+
+ain = ADC(Pin(26, Pin.IN, Pin.PULL_DOWN))
 cv1 = PWM(Pin(21))
+usb = Pin(24, Pin.IN)
 
 
 def wait_for_voltage(voltage):
+    wait_for_b1(0)
     if voltage != 0:
-        input(f"\nPlease plug {voltage}V into the CV input and then press Enter")
+        oled.centre_text(f'Plug {voltage}V into\nanalogue in\nthen press B1')
+        wait_for_b1(1)
     else:
-        input('\nPlease unplug all cables and then press Enter')
-    print('\n Calibrating...')
+        oled.centre_text(f'Please unplug\nall cables\nthen press B1')
+        wait_for_b1(1)
+    oled.centre_text('Calibrating...')
+    sleep(1.5)
     readings = []
     for reading in range(256):
-        readings.append(ain.pin.read_u16())
+        readings.append(ain.read_u16())
     return round(sum(readings)/256)
 
+def text_wait(text, wait):
+    oled.centre_text(text)
+    sleep(wait)
+    
+def fill_show(colour):
+    oled.fill(colour)
+    oled.show()
+    
+def flash(flashes, period):
+    for flash in range(flashes):
+        fill_show(1)
+        sleep(period/2)
+        fill_show(0)
+        sleep(period/2)
+    
+def wait_for_b1(value):
+    while b1.value() != value:
+        sleep(0.05)
+
+
+flash(8, 0.2)
+
+while usb.value() == 1:
```

Comment:
Consider adding a warning when usb detected instead of forcing users to unplug USB.

```suggestion

if usb.value() == 1:
    text_wait('Please connect\nrack power only\nnot USB', 4)
```